com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -395,6 +400,47 @@ class Job(_ResourceElement):
         >>> print (jobs[0].health)
         healthy
     """
+    def get_application_trace(self, path=None, name=None):
+        """Retrieves the application log and saves it to the specified path with the given name as a tar file.
+
+        If logs are retrieved with the same path and name as previously retrieved logs, the prior logs will be
+        overwritten.
+
+        Attributes:
+            path (str): a valid directory in which to save the application log output. Defaults to current dir.
+            name (str): the filename of the created tar file. Defaults to a name based on the job name.
+
+         Returns:
+            str: the path to the application logs tar file.
+         """
+        logger.debug("Retrieving application logs from: " + self.applicationLogTrace)
+        logs = self.rest_client.make_raw_request(self.applicationLogTrace)
+        
+        if name is None:
+            # Take the job name and remove colons (colons confuse the unix 'tar' command)
+            name = ''.join(self.name.split(':'))
+
+        name = name + "_app_logs.tar"
+        
+        if path is None:
+            path = os.getcwd()
+
+        path = path + "/" + name
+        try:
+            logfile = open(path, 'w+b')
+            logfile.write(logs.content)
```

Comment:
Is it possible to stream from the HTTP response into the file or does that do this?

Logs can be big and having to load it into memory first could be slow and problematic.

Reply:
It now streams the tar file contents using a 1K chunk size.